code/game/machinery/doors/firedoor.dm

```diff
@@ -188,30 +188,52 @@
 		for(var/obj/machinery/firealarm/fire_panel in place.firealarms)
 			fire_panel.set_status()
 
+/obj/machinery/door/firedoor/proc/CalculateWatchedTurfs()
+	watched_turfs = list()
+	for(var/dir in GLOB.cardinals)
+		var/turf/checked_turf = get_step(get_turf(src),dir)
+		watched_turfs |= checked_turf
+		RegisterSignal(checked_turf, COMSIG_TURF_EXPOSE, .proc/check_atmos)
+	watched_turfs |= get_turf(src)
```

Comment:
we will always want to call check_atmos at the end of CalculateWatchedTurfs right?

Reply:
CalculateWatchedTurfs is called in check_atmos

Reply:
but yes Ill do that